### ISSUE TITLE
HTTP CONNECT request sent in different packets while using an HTTPS Proxy 

### ISSUE THREAD

**Issue by brad-anton** [psf/requests#4884]
When using an HTTPS proxy, HTTP headers (and other parts of the request) are sent at different times then the initial HTTP `CONNECT` method, causing _some_ proxy services to abort the connection. 

## Expected Result

The entire HTTP request should be sent in the same packet as the `CONNECT` method. 

## Actual Result

HTTP Headers and trailing `\r\n` are sent at a different time. 

You can see this in the Wireshark Capture:

First Packet contains the HTTP CONNECT (from [here](https://github.com/python/cpython/blob/master/Lib/http/client.py#L895-L898)) 

![screen shot 2018-11-28 at 15 55 17](https://user-images.githubusercontent.com/1035793/49182352-10bf1600-f328-11e8-8f8b-8b78694ad125.png)

Second packet shows the left over `\r\n` (from [here](https://github.com/python/cpython/blob/master/Lib/http/client.py#L903))

![screen shot 2018-11-28 at 15 56 05](https://user-images.githubusercontent.com/1035793/49182338-06048100-f328-11e8-988b-b3268ec1d0e7.png)

This results in _some_ proxy servers aborting the connection.

## Reproduction Steps

```
>>> import requests
>>> proxies = { 'https': 'https://someproxy:443' }
>>> requests.get('https://www.google.com', proxies=proxies)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/user/Library/Python/2.7/lib/python/site-packages/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/Users/user/Library/Python/2.7/lib/python/site-packages/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/user/Library/Python/2.7/lib/python/site-packages/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/user/Library/Python/2.7/lib/python/site-packages/requests/sessions.py", line 596, in send
    r = adapter.send(request, **kwargs)
  File "/Users/user/Library/Python/2.7/lib/python/site-packages/requests/adapters.py", line 485, in send
    raise ProxyError(e, request=request)
requests.exceptions.ProxyError: HTTPSConnectionPool(host='www.google.com', port=443): Max retries exceeded with url: / (Caused by ProxyError('Cannot connect to proxy.', error(54, 'Connection reset by peer')))
```

## System Information
```
$ python -V
Python 2.7.11
```
(Same behavior on Python 3)

MacOSX

## Work Around
The following work around sends the entire HTTP `CONNECT` request at the same time:

https://gist.github.com/brad-anton/ff517025a86754793b680b3c5cfc9797

**Comment by sethmlarson** [comment]
This is an issue with urllib3, maybe we can use that new issue moving feature! :)

**Comment by nateprewitt** [comment]
@SethMichaelLarson I gave it a whirl but it looks like we can only move between repos in this org. @brad-anton would you mind making an issue in the urllib3 repo referencing this? Then we can close this out after. Thanks!

**Comment by sethmlarson** [comment]
Shame, I was looking forward to exchanging so many issues back and forth with our sister project. :)

**Comment by nateprewitt** [comment]
We’ll get there :) Incremental progress.

**Comment by sethmlarson** [comment]
This was [fixed in CPython 3.10](https://github.com/python/cpython/pull/24780) and thus in both urllib3 and Requests.
